protocols/identify/src/protocol.rs

```diff
@@ -236,14 +236,25 @@ impl TryFrom<structs_proto::Identify> for Info {
         let listen_addrs = {
             let mut addrs = Vec::new();
             for addr in msg.listen_addrs.into_iter() {
-                addrs.push(parse_multiaddr(addr)?);
+                match parse_multiaddr(addr) {
+                    Ok(a) => addrs.push(a),
+                    Err(e) => {
+                        debug!("Unable to parse multiaddr: {e:?}");
+                    }
+                }
             }
             addrs
         };
 
         let public_key = PublicKey::from_protobuf_encoding(&msg.public_key.unwrap_or_default())?;
 
-        let observed_addr = parse_multiaddr(msg.observed_addr.unwrap_or_default())?;
+        let observed_addr = match parse_multiaddr(msg.observed_addr.unwrap_or_default()) {
+            Ok(a) => a,
+            Err(e) => {
+                debug!("Unable to parse multiaddr: {e:?}");
+                Multiaddr::empty()
```

Comment:
This is not ideal but in my eyes not worse than the status quo. Downstream code has to handle a `Multiaddr::empty` due to the `unwrap_or_default` anyways.

One might first think that a remote could never return an observed address, i.e. an address that it observes the local node as, that the local node doesn't know. One would expect that the local node supports the multiaddr to all all its supported transport protocols. One exception is a relayed address though. The relay between A and B might use a protocol to B that A does not support nor be able to parse.

Reply:
> One might first think that a remote could never return an observed address, i.e. an address that it observes the local node as, that the local node doesn't know.

A remote can return all sorts of garbage :)

How big is the impact if we make this field an `Option`? That would be cleaner IMO.

Reply:
Agreed that `Option` would be the way to go here. That said, I think we should keep the patch release minimal, i.e. only do this on latest `master`.